src/app/campaign/store/models/flight.models.ts

```diff
@@ -52,15 +52,19 @@ export const getFlightZoneIds = (zones: any[]): string[] => {
   return zones.filter(z => z).map(z => z.id || z);
 };
 
-export const duplicateFlight = (flight: Flight, tempId: number): Flight => {
+export const duplicateFlightState = (flight: Flight, tempId: number, changed, valid): FlightState => {
```

Comment:
Ooh, also catching the previous form state when copying to a new form?  I'm guessing the new form looks pristine when you create it ... good catch!